### ISSUE TITLE
Proposal: a more typing-friendly experiment id pattern

### ISSUE THREAD

**Issue by ultmaster** [microsoft/nni#3102]
As proposed in the last meeting, I'm hoping to have a more typing-friendly experiment id pattern instead of the current "random-uppercase-lowercase-digits". We can use the concatenation of two short English words. Here are a few examples.

```
joint-drake        2020-11-18 01:17  (a day ago)
worthy-burro       2020-11-18 02:07  (a day ago)
safe-kitten        2020-11-18 02:15  (a day ago)
mint-lamb          2020-11-18 02:25  (a day ago)
fresh-mullet       2020-11-18 02:28  (a day ago)
amused-louse       2020-11-18 02:32  (a day ago)
cool-gannet        2020-11-18 02:34  (a day ago)
funny-tomcat       2020-11-18 09:42  (a day ago)
prime-rhino        2020-11-18 09:52  (a day ago)
eager-newt         2020-11-18 10:00  (a day ago)
crack-dragon       2020-11-18 10:47  (a day ago)
nearby-jennet      2020-11-18 11:08  (a day ago)
```

### Pros

* It's much easier for human to read and type an English words rather than something random.
* It saves the effort to use monospace font to display experiment IDs.

### Cons

* The collision likelyhood might be higher than before. Might not work on trials (because we can easily have thousands of trials). We can combine with some other patterns, such as digits or collision detection and retrying.

### Collision likelyhood analysis

According to the [reddit post](https://www.reddit.com/r/dataisbeautiful/comments/6jbt4d/a_distribution_of_english_words_by_length_using/), there are over 20,000 English words with length shorter than 6:

![image](https://user-images.githubusercontent.com/8463288/99628863-fa10ad00-2a71-11eb-8d8f-bb25f04b9d10.png)

So basically we have more than 400,000,000 combinations, according to birthday paradox, we need ~2836 experiments for a user to be 1% likely to have a collision. ~895 to be a 0.1% likely. This is almost impossible. Besides, we can just crash the experiments and retry if that actually happens.
